.ci/azure-pipelines.yml

```diff
@@ -12,38 +12,112 @@ pr:
     include:
     - '*'
 
-pool:
-  vmImage: 'ubuntu-latest'
-
-steps:
-- task: NodeTool@0
-  displayName: 'Install Node'
-  inputs:
-    versionSpec: '10.x'
-
-- script: 'npm ci'
-  displayName: 'Install Dependencies'
-
-- script: 'npx cordova prepare'
-  displayName: 'Prepare Cordova'
-
-- script: 'npx cordova build android --debug'
-  displayName: 'Build Debug'
-
-- script: 'npx cordova build android --release'
-  displayName: 'Build Release'
-
-- task: CopyFiles@2
-  displayName: 'Copy APKs'
-  inputs:
-    SourceFolder: 'platforms/android/app/build/outputs/apk/'
-    Contents: '**/*.apk'
-    TargetFolder: '$(Build.ArtifactStagingDirectory)'
-    flattenFolders: true
-
-- task: PublishBuildArtifacts@1
-  displayName: 'Publish APKs'
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-    ArtifactName: 'jellyfin-android'
-    publishLocation: 'Container'
+jobs:
+  - job: Build
+    displayName: 'Build'
+
+    strategy:
+      matrix:
+        Debug:
+          BuildConfiguration: debug
+          ArtifactSuffix: debug
+        Release:
+          BuildConfiguration: release
+          ArtifactSuffix: release
+        Libre:
+          BuildConfiguration: release
+          ArtifactSuffix: libre
+
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    steps:
+    - script: 'echo "##vso[task.setvariable variable=NODE_ENV]development"'
+      displayName: 'Set Node Variable'
+      condition: eq(variables['BuildConfiguration'], 'debug')
+
+    - script: 'echo "##vso[task.setvariable variable=TAG]$(git describe --tags)"'
+      displayName: 'Set Tag Variable'
```

Comment:
Why set these task variables? They seem like environment variables right? especially the first one might be better off in a variables section.
```yaml
variables:
  solution: '**/*.sln'
  buildPlatform: 'Any CPU'
  buildConfiguration: 'Release'
  yourEnvVar: '$(yourPipelineVariable)'
  yourOtherEnvVar: '$(yourOtherPipelineVariable)'
```

Reply:
The second one has to use a bash command to get the tag, which wouldn't work in the variable section right?

Reply:
Yes that is true. I wonder in the clone task doesn't have the output you need in it's output vars already.

Reply:
I can't seem to find any documentation for the clone step.

Reply:
I also can't find the source for the clone task, so I guess the current way is the only way.